textacy/fileio/utils.py

```diff
@@ -0,0 +1,203 @@
+    else:
+        f = io.open(filepath, mode=mode,
+                    encoding=encoding, errors=errors, newline=newline)
+
+    return f
+
+
+def make_dirs(filepath, mode):
+    """
+    If writing ``filepath`` to a directory that doesn't exist, all intermediate
+    directories will be created as needed.
+    """
+    head, _ = os.path.split(filepath)
+    if 'w' in mode and head and not os.path.exists(head):
+        os.makedirs(head)
+
+
+def split_content_and_metadata(items, content_field, itemwise=True):
+    """
+    Split content (text) from associated metadata, but keep them paired together,
+    for convenient loading into a ``TextDoc`` (with ``itemwise = True``) or
+    ``TextCorpus.from_texts()`` (with ``itemwise = False``). Output format depends
+    on the form of the input items (dicts vs. lists) and the value for ``itemwise``.
+
+    Args:
+        items (iterable(dict) or iterable(list)): an iterable of dicts, e.g. as
+            read from disk by :func:`read_json_lines() <textacy.fileio.read.read_json_lines>`,
+            or an iterable of lists, e.g. as streamed from a Wikipedia database dump::
+
+                >>> ([pageid, title, text] for pageid, title, text in
+                ...  textacy.corpora.wikipedia.get_plaintext_pages(<WIKIFILE>))
+
+        content_field (str or int): if str, key in each dict item whose value is
+            the item's content (text); if int, index of the value in each list item
+            corresponding to the item's content (text)
+        itemwise (bool, optional): if True, content + metadata are paired item-wise
+            as an iterable of (content, metadata) 2-tuples; if False, content +
+            metadata are paired by position in two parallel iterables in the form of
+            a (iterable(content), iterable(metadata)) 2-tuple
+
+    Returns:
+        generator(tuple(str, dict)): if ``itemwise`` is True and ``items`` is an
+            iterable of dicts; the first element in each tuple is the item's content,
+            the second element is its metadata as a dictionary
+        generator(tuple(str, list)): if ``itemwise`` is True and ``items`` is an
+            iterable of lists; the first element in each tuple is the item's content,
+            the second element is its metadata as a list
+        tuple(iterable(str), iterable(dict)): if ``itemwise`` is False and ``items``
+            is an iterable of dicts; the first element of the tuple is an iterable
+            of items' contents, the second is an iterable of their metadata dicts
+        tuple(iterable(str), iterable(list)): if ``itemwise`` is False and ``items``
+            is an iterable of lists; the first element of the tuple is an iterable
+            of items' contents, the second is an iterable of their metadata lists
+    """
+    if itemwise is True:
+        return ((item.pop(content_field), item) for item in items)
+    else:
+        return unzip(((item.pop(content_field), item) for item in items))
```

Comment:
ha! it's more complicated to explain than it is to code.

this reminds me: i meant to switch the default `itemwise=True` to `itemwise=False`.